### PR TITLE
Started alternate API for dictionaries

### PR DESCRIPTION
**Pull request by mwhelan**
Hey @robdmoore . I have converted half the API to have the category in between. I think it will work quite well but have not done the whole API until I've guaged feedback to see if it's the right direction to follow.

### CODE REVIEW

**Comment by mwhelan** [comment]
Properties get rid of the redundant () on each extension method. So, the choices of API so far are:

`Any.AddressAusCity
Any.AddressAus().City
Any.AddressAus.City`

**Comment by robdmoore** [comment]
I prefer property.

I'm comfortable with the ones that come with dossier being "special" since when you create ones for your own domain it makes sense you'd put them directly against Any.

**Comment by robdmoore** [comment]
I'd prefer that the final value is not a property though - it should be a method since it will return different things everytime.

File: TestStack.Dossier.Tests/EquivalenceClasses/AddressAusEquivalenceTests.cs [hunk 1]
```diff
@@ -22,24 +22,24 @@ protected override List<object[]> GetData()
             return new List<object[]>
             {
                 new object[]
-                {new Words(FromDictionary.AddressAusCity), GenerateTestCasesForSut(Any.AddressAusCity)},
+                {new Words(FromDictionary.AddressAusCity), GenerateTestCasesForSut(Any.AddressAus.City)},
```
Comment (robdmoore) [inline TestStack.Dossier.Tests/EquivalenceClasses/AddressAusEquivalenceTests.cs]: So this should be Any.AddressAus.City()

**Comment by robdmoore** [comment]
I'm happy with this PR.

**Comment by robdmoore** [comment]
@mwhelan I'm merging this so it can be released, but can you do another PR to talk about the `[Obsolete]` methods in the BREAKING_CHANGES file?

**Pull request merged**